### PR TITLE
EVG-14154: Remove option to set default patch parameters

### PR DESCRIPTION
**Pull request by hadjri**
[EVG-14154](https://jira.mongodb.org/browse/EVG-14154)

### Description 
The behavior of the "yes" flag when setting new default parameters for a patch had caused some misconception on the purpose of the flag.  Adding --yes in the CLI will mean that all confirmation (y/n) popups will be skipped, however this could be taken as "yes" meaning to actually answer (y) to all prompts which is not the case.  The flag "skip_confirm" has been introduced to be more explicit on the meaning of the flag and should be used by users in the future.  To keep backwards compatability, "yes" and "y" flags will still work to skip prompts, however "skip_confirm" is now the recommended flag.

Additionally, the option to set default parameters has been removed in order to prevent unwanted patch parameters to be forgotten about and persisted in the project.

### Testing 
Manual testing int the CLI confirming that "skip_confirm" flag exhibits the same behavior as "yes" and "y" to prevent confirmation popups, and that "skip_confirm" is available to see when the command "evergreen patch --help" is run.

### CODE REVIEW

**Review by ablack12 — CHANGES_REQUESTED** [review]
Rather than creating a PR which has all of the changes you made before, can you create a new PR off of the main branch? Then the diff will only include the actual changes you're introducing in this PR. You can do this either by deleting your existing EVG-14154 and then creating a new one from main, or creating a different branch (like EVG-14154-2) from main

**Comment by hadjri** [comment]
> Rather than creating a PR which has all of the changes you made before, can you create a new PR off of the main branch? Then the diff will only include the actual changes you're introducing in this PR. You can do this either by deleting your existing EVG-14154 and then creating a new one from main, or creating a different branch (like EVG-14154-2) from main

Branch is now up to date with main and only new changes are shown.

File: operations/patch_util.go [hunk 1]
```diff
@@ -347,20 +347,10 @@ func (p *patchParams) loadVariants(conf *ClientSettings) error {
...
-				p.Parameters, p.Project), false) {
-			conf.SetDefaultParameters(p.Project, p.Parameters)
-			if err := conf.Write(""); err != nil {
-				return err
-			}
-		}
-	} else {
-		p.Parameters = defaultParameters
-	}
+	p.Parameters = defaultParameters
```
Comment (ablack12) [inline operations/patch_util.go:353]: we can just set p.Parameters directly at 352 now

**Review by ablack12 — APPROVED** [review]
remember to increment the version string

**Pull request merged**